pages/admin/permit-holder/[permitHolderId].tsx

```diff
@@ -30,84 +30,106 @@ import {
-      );
-      setPreviousPhysicianData(
-        data.applicant.medicalHistory.map(record => ({
-          name: record.physician.name,
-          mspNumber: record.physician.mspNumber,
-          phone: record.physician.phone,
-        }))
-      );
-    },
-  });
+  const { data, refetch } = useQuery<GetPermitHolderResponse, GetPermitHolderRequest>(
+    GET_PERMIT_HOLDER,
+    {
+      variables: {
+        id: permitHolderId,
+      },
+      onCompleted: data => {
+        setPermits(
+          data.applicant.permits.map(permit => ({
+            rcdPermitId: permit.rcdPermitId,
+            expiryDate: permit.expiryDate,
+            applicationId: permit.applicationId,
+            isRenewal: permit.application.isRenewal,
+            status: permit.application.applicationProcessing.status,
+          }))
+        );
+        // TODO: uncomment when AWS is setup and we use real files
+        // const files: PermitHolderAttachedFile[] = [];
+        // data.applicant.fileHistory.forEach(application => {
+        //   application.documentUrls?.forEach(documentUrl => {
+        //     files.push({
+        //       appNumber: application.appNumber,
+        //       createdAt: application.createdAt,
+        //       fileUrl: documentUrl,
+        //     });
+        //   });
+        // });
+        // setAttachedFiles(files);
+        setMedicalHistoryData(
+          data.applicant.applications.map(application => ({
+            disability: application.disability,
+            createdAt: application.createdAt,
+            applicantApplication: application,
+          }))
+        );
+        setPreviousPhysicianData(
+          data.applicant.medicalHistory.map(record => ({
+            name: record.physician.name,
+            mspNumber: record.physician.mspNumber,
+            phone: record.physician.phone,
+          }))
+        );
+      },
+    }
+  );
 
   const toast = useToast();
 
+  // Submit updated medical information mutation
+  const [submitUpdatedMedicalInformation] = useMutation<
+    UpdateMedicalInformationResponse,
+    UpdateMedicalInformationRequest
+  >(UPDATE_MEDICAL_INFORMATION_MUTATION, {
+    onError: error => {
+      setIsErrorOnUpdateDoctor(true);
+      toast({
+        status: 'error',
+        description: error.message,
+      });
+    },
+  });
+
   // Submit edited doctor information mutation
   const [submitEditedDoctorInformation] = useMutation<
     UpsertPhysicianResponse,
     UpsertPhysicianRequest
   >(UPSERT_PHYSICIAN_MUTATION, {
     onCompleted: data => {
-      if (data.upsertPhysician.ok) {
-        toast({
-          status: 'success',
-          description: "Doctor's information has been edited.",
-        });
-      }
+      physicianId = data.upsertPhysician.physicianId;
     },
     onError: error => {
       toast({
         status: 'error',
         description: error.message,
       });
+      setIsErrorOnUpdateDoctor(true);
     },
-    refetchQueries: ['GetPermitHolder'],
```

Comment:
why was this removed? Do we not have to refetch `GetPermitHolder` anymore?

Reply:
In the case of the map number changing, we make two separate queries, and we would only want to refetch that query after the second one. Instead, we call `refetch()` at the end of `handleUpdateDoctorInformation` so that we only refetch once regardless of whether one or two queries were made.